### PR TITLE
[DI] Fixed custom services definition BC break introduced in ec7e70fb…

### PR DESCRIPTION
**Pull request by kiler129**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no, fixes previous BC
| Deprecations? | no
| Tests pass?   | yes (verified on Win only)
| Fixed tickets | #20608 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/ContainerBuilder.php [hunk 1]
```diff
@@ -840,8 +840,8 @@ public function findDefinition($id)
      */
     public function createService(Definition $definition, $id, $tryProxy = true)
     {
-        if ('Symfony\Component\DependencyInjection\Definition' !== get_class($definition)) {
-            throw new RuntimeException(sprintf('Constructing service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
+        if ($definition instanceof DefinitionDecorator) {
+            throw new RuntimeException(sprintf('Constructing decorated service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: this is not a decorated service. It is about using definition inheritance (yes, the name ``DefinitionDecorator`` is quite bad, but the service decoration feature was implemented years later)

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: Constructing service "foo" from parent definition "bar" etc.

Comment (kiler129) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: Oh, I didn't knew that and just assumed that by name. I'm thinking how we can actually handle that without breaking custom definition classes.

**Comment by kiler129** [comment]
The failure on appveyor doesn't seems to be related to this PR change.

File: src/Symfony/Component/DependencyInjection/ContainerBuilder.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: Can you update the message as suggested?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ContainerBuilder.php]: That's why I added the "parent" word above, which is also the vocabulary used to declare such services

**Comment by nicolas-grekas** [comment]
Thank you @kiler129.

**Comment by kiler129** [comment]
@nicolas-grekas: My pleasure ;)

**Pull request closed without merging**